frontend/app/repositories/post_repository.ts

```diff
@@ -0,0 +1,21 @@
+// postsRepository.ts
+import type {Post} from "~/entities/post";
+import axios from "axios";
+
+export class PostsRepository {
+    async get(): Promise<Post[]> {
+        // return [];
+        try {
+            const response = await axios.get<Post[]>('http://localhost:8080/api/posts', {
```

Comment:
I think we can store the api calls in a separate file. Or have separate files for each entity and its endpoints. I think it offers better readability and less repetition
Such a file might look like:

_./post_endpoints.ts_
 ```
                 const **BASE_URL** = 'http://localhost:8080/'
                  
                  export const API_ENDPOINTS = {
                       POSTS: `${**BASE_URL**}/posts`,
```
  

```suggestion
            const response = await axios.get<Post[]>(API_ENDPOINTS.POSTS, {
```